fxola/EPIC-mail
# #164380474 write tests for all endpoints, setup coveralls and deployment

#### What does this PR do?
- Implement tests, setup coveralls application deployment


#### Description of Task to be completed?
- write test cases for Auth(sign in and signup) endpoints
- write tests for messages endpoints
- update message service and controller to reflect LFA feedback
- install necessary dependencies for coveralls and coverage report
- setup coveralls and coverage report
- create travis yml file in preparation for integrating travis CI
- refactor message controller and router to follow DRY pattern
- refactor validation response to reflect LFA feedback
- update README file to follow LFA feedback convention
- setup build script and Procfile  for heroku deployment

#### How should this be manually tested?
- Clone the repository using ```git clone https://github.com/fxola/EPIC-mail.git .``` in your terminal
- run ```npm install``` on your terminal
- run ```npm run test``` to see test results
- run ```npm run coverage``` to see coverage report

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
- PT Story link - [#164380474] (https://www.pivotaltracker.com/n/projects/2314937/stories/164380474 )

#### Screenshots (if appropriate)
- N/A

#### Questions:
- N/A
